OpenSim/Simulation/SimbodyEngine/Body.cpp

```diff
@@ -167,42 +179,34 @@ void Body::addDisplayGeometry(const std::string &aGeometryFileName)
  */
 const SimTK::Inertia& Body::getInertia() const
 {
-	// Has not been set programmatically
-	if (_inertia.isNaN()){
-		// initialize from properties
-		const double& m = getMass();
-		// if mass is zero, non-zero inertia makes no sense
-		if (-SimTK::Eps <= m && m <= SimTK::Eps){
-			// force zero intertia
-			cout<<"Body '"<<getName()<<"' is massless but nonzero inertia provided.";
-			cout<<" Inertia reset to zero. "<<"Otherwise provide nonzero mass."<< endl;
-			_inertia = SimTK::Inertia(0);
-		}
-		else{
-			const SimTK::Vec6& Ivec = get_inertia();
-			try {
-				_inertia = SimTK::Inertia(Ivec.getSubVec<3>(0), Ivec.getSubVec<3>(3));
-			} 
-			catch (const std::exception& ex){
-				// Should throw an Exception but we have models we have released with
-				// bad intertias. E.g. early gait23 models had an error in the inertia
-				// of the toes Body. We cannot allow failures with our models so 
-				// raise a warning and do something sensible with the values at hand.
-				cout << "WARNING: Body " + getName() + " has invalid inertia. " << endl;
-				cout << ex.what() << endl;
-
-				// get some aggregate value for the inertia based on exsiting values
-				double diag = Ivec.getSubVec<3>(0).norm()/sqrt(3);
-
-				// and then assume a spherical shape.
-				_inertia = SimTK::Inertia(Vec3(diag), Vec3(0));
-				
-				cout << getName() << " Body's inertia being reset to:" << endl;
-				cout << _inertia << endl;
-			}
-		}
-	}
-	return _inertia;
+    // Has not been set programmatically
+    if (_inertia.isNaN()) {
+        const SimTK::Vec6& Ivec = get_inertia();
+        try {
+            const_cast<Body*>(this)->_inertia =
+                SimTK::Inertia(Ivec.getSubVec<3>(0), Ivec.getSubVec<3>(3));
+        }
+        catch (const std::exception& ex) {
+            // Should throw an Exception but we have models we have released
+            // with bad intertias. E.g. early gait23 models had an error in the
+            // inertia of the toes Body. We cannot allow failures with our
+            // models so raise a warning and do something sensible with the
+            // values at hand.
+            cout << "WARNING: Body " + getName() + " has invalid inertia. " << endl;
+            cout << ex.what() << endl;
+
+            // get some aggregate value for the inertia based on exsiting values
+            double diag = Ivec.getSubVec<3>(0).norm()/sqrt(3);
+
+            // and then assume a spherical shape.
+            const_cast<Body*>(this)->_inertia =
+                SimTK::Inertia(Vec3(diag), Vec3(0));
+
+            cout << getName() << " Body's inertia being reset to:" << endl;
+            cout << _inertia << endl;
```

Comment:
I meant that the setting of the internal `_inertia` should be done in finalizeFromProperties.